File: tools/official_import_config.py
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path
import os
import sys


ROOT = Path(__file__).resolve().parents[1]
DEFAULT_SLUG = "official-word-bank"


def read_arg(name: str, fallback: str = "") -> str:
    if name not in sys.argv:
        return fallback
    index = sys.argv.index(name)
    if index + 1 >= len(sys.argv):
        raise ValueError(f"{name} needs a value.")
    return sys.argv[index + 1]


def value_from_arg_or_env(arg_name: str, env_name: str, fallback: str) -> str:
    return read_arg(arg_name, os.environ.get(env_name, fallback))


@dataclass(frozen=True)
class ImportConfig:
    basename: str
    import_root: Path
    preview_prefix: str
    slug: str

    @property
    def raw_path(self) -> Path:
        return self.import_root / f"{self.basename}.raw.json"

    @property
    def review_queue_path(self) -> Path:
        return self.import_root / f"{self.basename}.review-queue.json"

    @property
    def reviewed_path(self) -> Path:
        return self.import_root / f"{self.basename}.reviewed.json"

    @property
    def reviewed_preview_path(self) -> Path:
        return self.import_root / f"{self.basename}.reviewed.preview.json"

    @property
    def merged_preview_path(self) -> Path:
        return self.import_root / f"words.{self.preview_prefix}-merged.preview.json"

    @property
    def source_manifest_path(self) -> Path:
        return self.import_root / "source-manifest.json"

    @property
    def review_queue_report_path(self) -> Path:
        return self.import_root / "review-queue-report.md"


def get_import_config() -> ImportConfig:
    slug = value_from_arg_or_env("--import-slug", "WORD_BANK_IMPORT_SLUG", DEFAULT_SLUG)
    basename = value_from_arg_or_env("--basename", "WORD_BANK_BASENAME", slug)
    import_root_text = value_from_arg_or_env("--import-root", "WORD_BANK_IMPORT_ROOT", "")
    import_root = (ROOT / import_root_text).resolve() if import_root_text else ROOT / "data-imports" / slug
    preview_prefix = "official" if slug == DEFAULT_SLUG and basename == DEFAULT_SLUG else basename
    return ImportConfig(basename=basename, import_root=import_root, preview_prefix=preview_prefix, slug=slug)

File: tools/official_import_config.js
const path = require("path");

const root = path.resolve(__dirname, "..");
const defaultSlug = "official-word-bank";

function readArg(name, fallback = "") {
  const index = process.argv.indexOf(name);
  if (index < 0 || index + 1 >= process.argv.length) {
    return fallback;
  }
  return process.argv[index + 1];
}

function valueFromArgOrEnv(argName, envName, fallback) {
  return readArg(argName, process.env[envName] || fallback);
}

function getImportConfig() {
  const slug = valueFromArgOrEnv("--import-slug", "WORD_BANK_IMPORT_SLUG", defaultSlug);
  const basename = valueFromArgOrEnv("--basename", "WORD_BANK_BASENAME", slug);
  const importRootText = valueFromArgOrEnv("--import-root", "WORD_BANK_IMPORT_ROOT", "");
  const importRoot = importRootText ? path.resolve(root, importRootText) : path.join(root, "data-imports", slug);
  const previewPrefix = slug === defaultSlug && basename === defaultSlug ? "official" : basename;
  return {
    basename,
    importRoot,
    root,
    slug,
    autoCandidatesPath: path.join(importRoot, `${basename}.auto-candidates.json`),
    autoCandidatesReportPath: path.join(importRoot, "auto-candidates-report.md"),
    providerRequestsPath: path.join(importRoot, `${basename}.provider-requests.jsonl`),
    rawPath: path.join(importRoot, `${basename}.raw.json`),
    reviewQueuePath: path.join(importRoot, `${basename}.review-queue.json`),
    reviewedDraftPath: path.join(importRoot, `${basename}.reviewed-draft.json`),
    reviewedDraftPreviewPath: path.join(importRoot, `words.${previewPrefix}-draft.preview.json`),
    reviewedDraftReportPath: path.join(importRoot, "reviewed-draft-report.md"),
    reviewedPath: path.join(importRoot, `${basename}.reviewed.json`),
    reviewedPreviewPath: path.join(importRoot, `${basename}.reviewed.preview.json`),
    mergedPreviewPath: path.join(importRoot, `words.${previewPrefix}-merged.preview.json`)
  };
}

module.exports = {
  getImportConfig,
  readArg,
  root
};
